provider/testing/provider.go

```diff
@@ -0,0 +1,28 @@
+package providertesting
```

Comment:
Please register new mocks in `.mockery.yml` and run `make generate` instead of writing them manually.
You can easily add `helper.go` to add nicer stuff if needed (see the other existing mocks).